### PR TITLE
Non visual properties

### PR DESCRIPTION
**Pull request by ggetz**
Added `meta` property to `Cesium3DTileStyle` to evaluate non-visual properties.

ex.
```
"style" : {
    "meta" : {
        "description" : "Hello, ${FeatureName}"
    }
}
```

```
var str = style.meta.description.evaluate(feature);
```

If meta is not supplied, or something other than the meta properties are supplied, (such as "meta" : 1), `meta` defaults to `{}`.

Added tests and updated sandcastle example.

### CODE REVIEW

**Comment by pjcozzi** [comment]
> If meta is not supplied, or something other than the meta properties are supplied, (such as "meta" : 1), meta defaults to {}.

In general, don't "fix" bad input like this.

`undefined` can default to `{}`

`1` is invalid.

File: Source/Scene/Cesium3DTileStyle.js [hunk 1]
```diff
@@ -80,6 +83,18 @@ define([
 
         that._show = show;
 
+        var meta = {};
+        var metaJson = styleJson.meta;
+        if(defined(metaJson)) {
```
Comment (pjcozzi) [inline Source/Scene/Cesium3DTileStyle.js]: Whitespace after `if`.

File: Source/Scene/Cesium3DTileStyle.js [hunk 2]
```diff
@@ -80,6 +83,18 @@ define([
...
+            for (var property in metaJson) {
+                if (metaJson.hasOwnProperty(property)) {
+                    meta[property] = new Expression(styleEngine, metaJson[property]);
+                }
+            }
+        }
+
+        that.meta = meta;
```
Comment (pjcozzi) [inline Source/Scene/Cesium3DTileStyle.js]: `that._meta`, right?

**Comment by pjcozzi** [comment]
Can you also open a PR to update the 3D Tiles spec?

**Comment by ggetz** [comment]
> 1 is invalid.

So throw a developer error?

**Comment by pjcozzi** [comment]
Until we have full validation worked out, if it is defined, just assume it is an object like we do elsewhere.

**Comment by ggetz** [comment]
OK

**Comment by ggetz** [comment]
OK, I made the updates and also added the examples I used in the spec to the tests.

**Comment by pjcozzi** [comment]
Looks good, open a new PR for any new edge cases we find in https://github.com/AnalyticalGraphicsInc/3d-tiles/pull/85.
